apps/ejabberd/src/mod_vcard_ldap.erl

```diff
@@ -368,17 +373,22 @@ search_internal(State, Data) ->
     Limit = State#state.matches,
     ReportedAttrs = State#state.search_reported_attrs,
     Op = State#state.search_operator,
-    Filter = eldap:'and'([SearchFilter,
-                          eldap_utils:make_filter(Data, UIDs, Op)]),
-    case eldap_pool:search(EldapID,
-                           [{base, Base}, {filter, Filter}, {limit, Limit},
-                            {deref_aliases, State#state.deref_aliases},
-                            {attributes, ReportedAttrs}])
-        of
-      #eldap_search_result{entries = E} ->
-          search_items(E, State);
-      _ -> error
-    end.
+    Filter = eldap:'and'([SearchFilter, eldap_utils:make_filter(Data, UIDs, Op)]),
+    E = eldap_pool_search(EldapID, Base, Filter, State#state.deref, ReportedAttrs, error),
+    case E of
+      error ->
+        error;
+      E ->
+        Limited = limited_results(E, Limit),
+        search_items(Limited, State)
+     end.
+
+limited_results(E, Limit) when length(E) > Limit ->
+  lists:nthtail(Limit, E);
```

Comment:
Are you sure this is correct? If there are 12 results with limit set to 10, this call will result in 2-element list, which is not exactly what we want, right?